app/MMB/src/scenes/timeline/TimelineEvent.js

```diff
@@ -0,0 +1,53 @@
+// @flow
+
+import * as React from 'react';
+import { View } from 'react-native';
+import { StyleSheet, Color, Text } from '@kiwicom/mobile-shared';
+
+import TimelineRow from './TimelineRow';
+
+type Props = {|
+  +dateLocation: ?React.Element<typeof View>,
+  +icon: React.Element<typeof View>,
+  +mainContent: React.Element<typeof View | typeof Text>,
```

Comment:
Technically this code is correct. But I believe it's better not to create these restrictions. Just `React.Node` is enough - do not overcomplicate it... :)

It's clever to do it for translations for example where you want to be sure not to forget. But in this case, you can really accept "any" element. Maybe the `icon` should be type of `Icon` (?). But `React.Node` should be OK for most of the cases.

Reply:
I understand, this is indeed much simpler :)
I renamed the `icon` props as it is slightly misleading: it is not just an icon, it is an icon surrounded by vertical lines.